### ISSUE TITLE
Crashing on Linux. 

### ISSUE THREAD

**Issue by isalcedo** [avocode/avocode#2109]
I'm using the latest version of avocode on Archlinux.

From two versions ago always when I run Avocode directly it crash giving me the option to send and issue, restart or quit. 

If I quit and next try to run it directly again, the crash happen, only if I chose to restart Avocode run without crash but I lost my login. I'm forced to login again.

This is kinda tedious.
---
* **Version**: 3.3.2
* **Web/Desktop**: Desktop LXDE
* **Platform**: linux (Archlinux)
---
```
{}
```

**Comment by jakubzitny** [comment]
Hi, thanks a lot for this report. We're already aware of this problem and we're trying to find the root cause of this behaviour but we're not really able to reproduce it properly.

Can you please try using this beta build of Avocode to see if the problems go away?
-> https://we.tl/R64ZldyRqh

The build should definitely help, we'll be happy if you use it and give us a feedback.

**Comment by isalcedo** [comment]
@jakubzitny Hey, sorry for the **strong** text on "This is kinda tedious.", Was not my intention. I don't know also why that text is on bold, I didn't activated it. D:

**Comment by jakubzitny** [comment]
No problem 😉 it's because of the line below it. Did you have a chance to try out the other build I shared?

**Comment by isalcedo** [comment]
@jakubzitny yeah, at this moment. Bugs keeps happening. Is because the recent electron update in Archlinux?

**Comment by jakubzitny** [comment]
I see, okay, thanks a lot for trying. We had a hypothesis that this is related to one other bug, that we're already fixing right now, but it's not.

What electron update in Arch do you mean?

We already heard about similar behaviour from at least one other Linux user in Avocode Community Slack channel. We'll be trying to fix the problem, but we are not able to reproduce the bug on our Linux machines.

Please bear with us, if this really breaks your workflow, you can use our web app at [app.avocode.com](https://app.avocode.com).

**Comment by isalcedo** [comment]
I actually don't know. My electron version in Archlinux is v2.0.5. I said that because Archlinux is Rolling Release, you know, some times the updates break things.

**Comment by jakubzitny** [comment]
But we're bundling builtin Electron in Avocode. Do you think you actually get the Arch's Electron when launching Avocode? Can you check it please?

- press `alt` and go to `Help -> Troubleshooting -> Open Dev Tools` menu
- type `process.versions` and enter

and send us the output, please.

**Comment by jakubzitny** [comment]
And please, do it for both the official release and this one I shared above.

**Comment by isalcedo** [comment]
`Object {http_parser: "2.7.0", node: "7.9.0", v8: "5.8.283.38", uv: "1.11.0", zlib: "1.2.11"…}ares: "1.10.1-DEV"atom-shell: "1.7.5"chrome: "58.0.3029.110"electron: "1.7.5"http_parser: "2.7.0"modules: "54"node: "7.9.0"openssl: "1.0.2k"uv: "1.11.0"v8: "5.8.283.38"zlib: "1.2.11"Symbol(util.inspect.custom): (depth) => {…}arguments: (...)caller: (...)length: 1name: ""__proto__: function ()[[FunctionLocation]]: util.js:1083__proto__: Object`

![avocode error](https://user-images.githubusercontent.com/2588745/43653359-6c1b9720-970d-11e8-946e-bb1a1e96c06c.png)

Current Stable Version

**Comment by jakubzitny** [comment]
Yeah, it's not using Arch's electron. Thanks a lot, we'll be trying to investigate this further. 🙂

**Comment by isalcedo** [comment]
Whay about the other warnings?

**Comment by jakubzitny** [comment]
That's just saying there is a problem with `~/.avocode` which is also related to logging out thing. But it doesn't say us a lot more.

**Comment by isalcedo** [comment]
Hey dude. Some news... I don't know how to exactly say this, I'll do my best with my poor English.

You know, when I open up Avocode it's shows the restart message, well, if I let Avocode restart without any interaction, the session is closed, it force me to login again, **BUT!** if I interact with Avocode opening a Folder while its reconnecting, it reconnect successfully and the session holds.

I don't know if that helps...

**Comment by bolekkerous** [comment]
Hey @isalcedo -
we've made a special build with a possible fix for this problem. Could you please try downloading it and trying out whether it helps or not?

https://we.tl/LAgahdBKL7

Thanks!

**Comment by isalcedo** [comment]
Same. 

Warnings:
![captura de pantalla de 2018-08-16 08-46-38](https://user-images.githubusercontent.com/2588745/44212443-04280080-a131-11e8-9079-39fe96af6c33.png)
Errors:
![captura de pantalla de 2018-08-16 08-46-49](https://user-images.githubusercontent.com/2588745/44212454-0b4f0e80-a131-11e8-8aaf-49390bb51eac.png)

**Comment by jakubzitny** [comment]
Could you also send us a screenshot of the console, with `Verbose` mode turned on, please @isalcedo? Not just errors and warnings, these are not helpful in this case.

Thanks a lot for helping.

**Comment by isalcedo** [comment]
![debug-1](https://user-images.githubusercontent.com/2588745/44264530-8fb09880-a1e8-11e8-8e0a-713c29b5058b.png)
![debug-2](https://user-images.githubusercontent.com/2588745/44264531-8fb09880-a1e8-11e8-8050-17d6fad39cdc.png)
![debug-3](https://user-images.githubusercontent.com/2588745/44264533-8fb09880-a1e8-11e8-8a14-690a126a9e99.png)
![debug-4](https://user-images.githubusercontent.com/2588745/44264534-90492f00-a1e8-11e8-8821-28ed24202b31.png)

^ Same things, scrolling awhile. Trust me is the same. I must think this is a repetitive cycle, of course is not a infinite loop, at some point it ends.

Next:
![debug-5](https://user-images.githubusercontent.com/2588745/44264535-90492f00-a1e8-11e8-971e-be7c24396aa6.png)
![debug-6](https://user-images.githubusercontent.com/2588745/44264536-90492f00-a1e8-11e8-8ff9-d4819e6fcd01.png)

**Comment by jankuca** [comment]
Hi @isalcedo, I will actually probably need the whole log to reconstruct the situation. Can you please save it and e-mail me the file at jankuca@avocode.com?

<img width="272" alt="screen shot 2018-08-17 at 13 47 26" src="https://user-images.githubusercontent.com/367262/44264794-35352d80-a224-11e8-8ae1-871f13a85147.png">

**Comment by isalcedo** [comment]
Hey Guys, I'm not using Gnome or KDE. (no enough RAM, tell me if you know about a sale for DDR3 Modules :grin: ) Some times some apps crash because the lack of policykit or some keyrings. I don't know if is the case.

I'm on LXDE with Openbox at minimum.

OMW @jankuca

**Comment by isalcedo** [comment]
Is here for the community and in your Email @jankuca.

[AvocodeSpecial-3.4-Build-Bug-GitHub-Issue-2109-08-17-2018___1534506630224.log](https://github.com/avocode/avocode/files/2297070/AvocodeSpecial-3.4-Build-Bug-GitHub-Issue-2109-08-17-2018___1534506630224.log)

**Comment by blakesisu** [comment]
Hi there, I'm on arch (manjaro) and my build that I just downloaded today from the arch repo is consistently crashing. It crashed on first run, then it worked great for me, then I closed and attempted to reopen it a few hours later and it keeps crashing on first load.

**Comment by jakubzitny** [comment]
Yeah, thanks for reporting it @blakesisu, it's probably the same problem. A number of Linux users is complaining about it. We're trying to find the right fix, however, we're not able to reproduce it on any of our Linux machines. For now we recommend using the web version (in Chrome/Chromium) or hacking around (as it's mentioned above in the thread) to make it work during every start. We'll update you here as soon as we have an improvement.

**Comment by isalcedo** [comment]
Arch updated Electron. Seems fixed. I'll do some another tests. Maybe was related to system electron all the time.

**Comment by jakubzitny** [comment]
I see, thanks for the update @isalcedo. Do you use Avocode with the Electron from Arch normally or is it more of a special case? I know most of Arch users are fluent in switching the deps, not sure how much of an effort is it for you.

**Comment by isalcedo** [comment]
@jakubzitny I usually don't change anything by myself but Avocode is installed from AUR, an user maintained repository for custom builds, maybe this links can help you in the investigation.

https://aur.archlinux.org/packages/avocode/
https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=avocode

**Comment by jakubzitny** [comment]
I know, we're actually maintaining AUR. But in the Avocode package in AUR, there should be Electron  already prepacked, but you're saying Arch updated Electron, which would mean you're using the Arch's Electron instead. How are you launching Avocode, please?

**Comment by isalcedo** [comment]
I actually don't know. I just installed Avocode from AUR, the bug was happening, that day after an update (electron was listed as one package in the update list) Avocode stop crashing. I don't know some actually good reason.

**Comment by isalcedo** [comment]
Talking about the devil, in Aur:

~~ looking for new pkgbuilds and fetching them...
==> ERROR: source_x86_64 must be an «array»
2018-09-12 12:42:22,130 - wrappers - makepkg - ERROR - makepkg query ['makepkg', '--printsrcinfo']

**Comment by jakubzitny** [comment]
Right, thanks for pointing it out @isalcedo, I'll fix it right away.

**Comment by jakubzitny** [comment]
Done, can you please try it one more time @isalcedo?

**Comment by Grafikart** [comment]
Dunno if it helps but can the problem come from Visual Studio Code which require Electron from the repository ? Could this Electron installation conflicts with avocode ? 

(I'm on arch linux)

**Comment by jakubzitny** [comment]
It should not, but it's possible, how are you launching Avocode, please @Grafikart?

**Comment by Grafikart** [comment]
I launched it using the aur package (/bin/avocode)
But I also tried to launch it using the .zip from avocode website.

In both cases I tried to remove `~/.config/Avocode` and `~/.avocode` before launching

**Comment by jakubzitny** [comment]
Thanks for the info @Grafikart. Would you be willing to try it without the VSCode installation or clearing the other Electron somehow? I doubt that it is conflicting, but just to be 100% sure I'd try it.

If you don't have time to debug this, you can use our web version in Chrome/Chromium, that one should work properly.

**Comment by Grafikart** [comment]
@jakubzitny I tried to remove electron but I got the same problem, is there a way to get the console to give you more details ?

**Comment by jakubzitny** [comment]
Press `alt` and go to `Help -> Troubleshooting -> Open Dev Tools` menu.

Or launch the app with `-- --dev-tools` parameter from CLI.

**Comment by Grafikart** [comment]
I'm using the zip downloaded from Avocode to run my application

```
/home/xxxxxxx/App/Avocode/resources/app.asar/node_modules/@avocode/stage/node_modules/raven-js/src…:35 AuthManager#initAuth() -> auth error: 401 Error: No token data
    at e.value (avocode.min.js:1)
    at t.value (avocode.min.js:1)
    at t.value (avocode.min.js:1)
    at e.value (avocode.min.js:1)
    at Object.<anonymous> (avocode.min.js:1)
    at i (avocode.min.js:1)
    at Object.<anonymous> (avocode.min.js:1)
    at i (avocode.min.js:1)
    at avocode.min.js:1
    at avocode.min.js:1
console.(anonymous function) @ /home/xxxxxxx/App/Avocode/resources/app.asar/node_modules/@avocode/stage/node_modules/raven-js/src…:35
/home/xxxxxxx/App/Avocode/resources/app.asar/node_modules/@avocode/stage/node_modules/raven-js/src…:35 (node:11576) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 SWAP listeners added. Use emitter.setMaxListeners() to increase limit
console.(anonymous function) @ /home/xxxxxxx/App/Avocode/resources/app.asar/node_modules/@avocode/stage/node_modules/raven-js/src…:35
/home/xxxxxxx/App/Avocode/resources/app.asar/node_modules/@avocode/stage/node_modules/raven-js/src…:35 (node:11576) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 change listeners added. Use emitter.setMaxListeners() to increase limit
console.(anonymous function) @ /home/xxxxxxx/App/Avocode/resources/app.asar/node_modules/@avocode/stage/node_modules/raven-js/src…:35
/home/xxxxxxx/App/Avocode/resources/app.asar/node_modules/@avocode/stage/node_modules/raven-js/src…:35 AuthStateIo#invalidate() -> could not delete .token file: Error: ENOENT: no such file or directory, unlink '/home/xxxxxxx/.avocode/.token'
console.(anonymous function) @ /home/xxxxxxx/App/Avocode/resources/app.asar/node_modules/@avocode/stage/node_modules/raven-js/src…:35
api-iam.intercom.io/messenger/web/ping Failed to load resource: the server responded with a status of 400 ()
```

**Comment by jakubzitny** [comment]
Thanks, but this does not really help us, these are unrelated warnings actually. We'll try to do more investigation on our side and get back to you. We'll also be upgrading Electron to the latest version in `3.6.0` so that might also help.

Would you be ok to use the web version in the meantime?

**Comment by Grafikart** [comment]
@jakubzitny yeah no problem i'm currently using the web version.

**Comment by Grafikart** [comment]
btw the problem is very inconsistent. Without changing things sometimes it opens on a blank black window (without the top menu), sometimes you get the menu but not the rest of the application, and sometimes it just work. I'll keep you informed if I find a way to reproduce the problem easily.

**Comment by jakubzitny** [comment]
Yeah, we've heard that from another Arch user already. Thanks a lot. :)

**Comment by mearaj** [comment]
I was facing the same issue. It's solved now. I played around with lot of things to fix it. I even tried to start the app from terminal and I was getting out of memory error. I am not sure what resolved my issue but I believe that the most probable cause of my issue was ulimit(open file limits of user). I increased my ulimit in /etc/security/limits.conf. This is the link to better understand it
https://wiki.archlinux.org/index.php/Limits.conf
I hope that it may help.Also don't forget to restart your pc in order to get changes into effect.
Thanks.

**Comment by sakralbar** [comment]
I can confirm that changing the limits.conf file solves the problem for me in Debian buster.
I added this lines to my limits.conf:
```
*         hard    nofile      500000
*         soft    nofile      500000
root      hard    nofile      500000
root      soft    nofile      500000
```
